fix: remove unuseful patch message from instrumentations (#2161)
open-telemetry/opentelemetry-js-contrib: GUIDELINES.md

    };
    ...
    ```

> ## Diag Channel

Reviewer: 🤩  Love having this guideline written out here

> ## Diag Channel

Author: I remember that "diag channel" is how we used to refer to it when speaking about it, but I now understand it was not formal name. 
Changed to `Diag Logging`. Thanks!

- The instrumentation is not auto loaded - due to issue with the require/import interception, an unsupported version of the instrumented package, or some other issue. This knowledge can pin-point the issue to the instrumentation package.
- The instrumentation patch was applied but expected spans are missing - this can spot the light on the instrumented package logic, configuration, limits, otel sdk, or other issues.

It can also be useful to know when the instrumentation is loaded and patched, to understand the order of operations in the application.

Instrumentation packages should use the `@opentelemetry/instrumentation` package `BaseInstrumentation` class to register patches and unpatch callbacks for specific require/import of the instrumented package, it's dependency or an internal module file. When this mechanism is used, the base class will automatically emit a debug message on instrumentation diag component logger, looking like this:

```shell
@opentelemetry/instrumentation-foo Applying instrumentation patch for module on require hook {
  module: 'foo',
  version: '1.2.3',

>   baseDir: '<your directory>/node_modules/mongoose'

Reviewer: ```suggestion
  baseDir: '<your directory>/node_modules/foo'
```
nit: for consistency match module name

}
```

Instrumentation should not add additional debug messages for triggering the patching and unpatching callbacks, as the base class will handle this.

> Instrumentation may add additional patch/unpatch messages for specific functions if it is expected to help in troubleshooting issues with the instrumentation. Few examples:

Reviewer: Oh, another thought I had that I think will probably be updated in the follow-up PR that replaces inconsistent usages of `api.diag.debug` and similar that aren't using the instrumentation `this` to allow for `this._diag.debug`. When we make that change we should update guidelines for that as well, though it will already be easier to follow suit once these are all more consistent 😄

Author: certainly, I aim to first remove or hoist unneeded diag messages, and then will transform all remaining diag message to use `this._diag` and update the GUIDELINES.md 👍 

https://github.com/open-telemetry/opentelemetry-js/pull/4663 will also help in hosting many diag prints and remove the repeated code from  dozens of places in contrib.

> - If the patch logic is conditional, and user can benefit from ensuring the condition is met and the patch happened. `koa` patching logic examine an object and branch between patching it as router vs middleware, which is applied at runtime. `awslambda` will aboard patching if the environment is not configured properly.

Reviewer: ```suggestion
- If the patch logic is conditional, and user can benefit from ensuring the condition is met and the patch happened. `koa` patching logic examine an object and branch between patching it as router vs middleware, which is applied at runtime. `aws-lambda` will abort patching if the environment is not configured properly.
```
Let me know if I have that right, I believe the intent is to say it will not patch, so abort is the word we want here.

> - When the patch is not applied directly on a `moduleExports` object in the `BaseInstrumentation` callbacks, but rather from an event in the package, like creating new client instance, registering a listener, etc. `fastify` instrumentation applies a patch when an hook is added to the fastify app instance, which is patched from `moduleExports`.

Reviewer: ```suggestion
- When the patch is not applied directly on a `moduleExports` object in the `BaseInstrumentation` callbacks, but rather from an event in the package, like creating new client instance, registering a listener, etc. `fastify` instrumentation applies a patch when a hook is added to the fastify app instance, which is patched from `moduleExports`.
```

> - In situations where the patch logic is not trivial and it helps to specify patch events in the right context and nuances. `aws-lambda` logs additional properties extracted from the labmda framework and exposes them for troubleshooting.

Reviewer: ```suggestion
- In situations where the patch logic is not trivial and it helps to specify patch events in the right context and nuances. `aws-lambda` logs additional properties extracted from the lambda framework and exposes them for troubleshooting.
```

> The cases above are not covered by the base class and offer additional context to the user trouble shooting an issue with the instrumentation.

Reviewer: ```suggestion
The cases above are not covered by the base class and offer additional context to the user troubleshooting an issue with the instrumentation.
```
